include/mostly_harmless/utils/mostlyharmless_Timer.h

```diff
@@ -12,8 +12,20 @@ namespace mostly_harmless::utils {
      *
      * Runs on its own thread - so thread safety concerns all apply here.
      */
-    class Timer {
+    class Timer : public std::enable_shared_from_this<Timer> {
+    private:
+        struct Private {
+            explicit Private() = default;
+        };
+
     public:
+        // constructor is only usable by this class
+        explicit Timer(Private) {}
```

Comment:
I'm not the biggest fan of this pattern, but apparently it's idiomatic C++, according to [cppreference](https://en.cppreference.com/w/cpp/memory/enable_shared_from_this)